### PR TITLE
feat: introduce workflow for doing a release to production

### PR DESCRIPTION
**Pull request by levivannoort**
Close: #4966 

## PR Details

**Blocker**: this pull request requires that pull request #5021 is merged first and also includes the name suggested in the comments, namely `nx.template.yml` instead of `nx-affected.template.yml`.

This pull request introduces a new workflow that is meant to bridge the gap between the implementation of argocd image update and the current implementation after migration (i.e. manual changes on the production manifests). It first determines the previous git tag/release based on the one that triggered the workflow, after which it check which apps are affected and deployed in the form of container (i.e. provides output `affected-dockerable`). These affected components are then packaged and pushed to a container registry, and the last jobs takes care of changing the image tag of the affected components and subsequently creates a pr on the manifest repository.

## PR Checklist

- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

**Comment by levivannoort** [comment]
example of workflow output: https://github.com/amplication/amplication-manifests/pull/30

**Review by overbit — APPROVED** [review]

File: .github/workflows/deployment.production.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
+name: Deployment production
+
+on:
+  release:
+    types: [published]
+
```
Comment (overbit) [inline .github/workflows/deployment.production.yml:6]: Let's add the NX_CLOUD_ACCESS_TOKEN variable to leverage Nx Cloud distributed cache

**Review by overbit — CHANGES_REQUESTED** [review]

File: .github/workflows/deployment.production.yml (hunk 1 above)
Comment (levivannoort) [reply .github/workflows/deployment.production.yml:6]: add the token as an environments variable

**Review by overbit — APPROVED** [review]

**Review by overbit — APPROVED** [review]
💯  finally automated production releases!

**Pull request merged**